modules/apstreamBidAdapter.js

```diff
@@ -0,0 +1,481 @@
+      host: match[4] || '',
+      hostname: match[4] ? match[4].split(':')[0] : '',
+      pathname: match[5] || '',
+      search: match[7] || '',
+      hash: match[9] || '',
+      toString: function () {
+        return url;
+      }
+    };
+
+    res.origin = res.scheme + '://' + res.host;
+    return res;
+  }
+
+  function location() {
+    var url = overwrite || window.location.toString();
+    url = url.replace(/\.demo\.audienceproject\.com\//, '/');
+
+    if (cache.url === url) {
+      return cache.parsed;
+    }
+    var parsed = parseUrl(url);
+    cache.url = url;
+    cache.parsed = parsed;
+    return parsed;
+  }
+
+  function getDaysSinceApEpoch() {
+    var timeDiff = (new Date()).getTime() - (new Date(2019, 0, 1)).getTime();
+    var daysSinceApEpoch = Math.floor(timeDiff / (1000 * 3600 * 24));
+    return daysSinceApEpoch;
+  }
+
+  function generateDsu() {
+    var dsuId = utils.generateUUID();
+    var loc = location();
+
+    var dsuIdSuffix = hashWithKey(dsuId + loc.toString());
+    var suffix4 = dsuIdSuffix.substr(0, 4);
+    var suffix8 = dsuIdSuffix.substr(4);
+
+    dsuId = dsuId.substr(0, 19) + suffix4 + '-' + suffix8;
+
+    var daysSinceApEpoch = getDaysSinceApEpoch();
+    var originHash = hashWithKey(loc.origin);
+
+    var metadata = [
+      DSU_CREATOR.USERREPORT,
+      daysSinceApEpoch,
+      originHash
+    ].join('.');
+    var signature = hashWithKey(dsuId + metadata + SIGNATURE_SALT);
+
+    return [DSU_VERSION_NUMBER, signature, dsuId, metadata].join('.');
+  }
+
+  function readOrCreateDsu() {
+    var dsu;
+    try {
+      dsu = storage.getDataFromLocalStorage(DSU_KEY);
+    } catch (err) {
+      return null;
+    }
+
+    if (!dsu) {
+      dsu = generateDsu();
+    }
+
+    try {
+      storage.setDataInLocalStorage(DSU_KEY, dsu);
+    } catch (err) {
+      return null;
+    }
+
+    return dsu;
+  }
+
+  return {
+    readOrCreateDsu: readOrCreateDsu
+  }
+})();
+
+function serializeSizes(sizes) {
+  if (Array.isArray(sizes[0]) === false) {
+    sizes = [sizes];
+  }
+
+  return sizes.map(s => s[0] + 'x' + s[1]).join('_');
+}
+
+function getRawConsentString(gdprConsentConfig) {
+  if (!gdprConsentConfig || gdprConsentConfig.gdprApplies === false) {
+    return null;
+  }
+
+  return gdprConsentConfig.consentString;
+}
+
+function getConsentStringFromPrebid(gdprConsentConfig) {
+  const consentString = getRawConsentString(gdprConsentConfig);
+  if (!consentString) {
+    return null;
+  }
+
+  let isIab = config.getConfig('consentManagement.cmpApi') != 'static';
+  let vendorConsents = (
+    gdprConsentConfig.vendorData.vendorConsents ||
+    (gdprConsentConfig.vendorData.vendor || {}).consents ||
+    {}
+  );
+  let isConsentGiven = !!vendorConsents[CONSTANTS.GVLID.toString(10)];
+
+  return isIab && isConsentGiven ? consentString : null;
+}
+
+function getIabConsentString(bidderRequest) {
+  if (utils.deepAccess(bidderRequest, 'gdprConsent')) {
```

Comment:
hey @frstua , in case someone did not include `consentManagement` module in their PBJS build or they simply didn't enable it (for example, they know they're not in GDPR zone so no need to enable it), `gdprConsent` will be undefined. This function will then return `null` and you will skip setting the `dsu`. Is this intentional for some reason?

Reply:
Hi @aleksatr. I changed logic which works with dsu and consent. But now the tests are failed by reason not related to my adapter.

Reply:
Hi @aleksatr. Good catch, I changed logic which sets dsu.